### ISSUE TITLE
Add Google Trace propagator

### ISSUE THREAD

**Issue by lucacasonato** [open-telemetry/opentelemetry-rust#370]
Should this be added to `opentelemetry` or `opentelemetry-contrib`? Unless anyone else wants to implement this, I will do it.

Go implementation for opencensus: https://github.com/census-instrumentation/opencensus-go/blob/v0.22.5/exporter/stackdriver/propagation/http.go

**Comment by jtescher** [comment]
I think @djc was moving some of https://github.com/vivint-smarthome/opentelemetry-stackdriver over, could work out what the best path forward is for that. I think either way the contrib crate is probably the best place for it currently.

**Comment by TommyCpp** [comment]
I think https://github.com/vivint-smarthome/opentelemetry-stackdriver only have `exporter`. If that's the case, then I don't know anyone else is working on the propagator.

**Comment by jtescher** [comment]
@TommyCpp good point, can be done separately if they only have an exporter. Looks like contrib is still the right place for the propagator either way according to [the propagators distribution spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/context/api-propagators.md#propagators-distribution).

**Comment by djc** [comment]
@jtescher I still don't like the idea of throwing a bunch of otherwise unrelated code in a single contrib crate; it would probably need complex feature combinations to make sure dependencies match your particular needs at any time. What's the argument against doing a top-level opentelemetry-google-cloud-ops crate (to be bikeshedded) containing both an exporter, a propagater, and whatever else makes sense for that particular environment?

(Alternatively, why aren't opentelemetry-jaeger, opentelemetry-zipkin etc all in the contrib crate?)

**Comment by jtescher** [comment]
@djc I think having a top level crate works as well for common exporters like aws/gcp, having more niche exporters like newrelic/datadog at the top level could become a maintenance burden. Was mentioning contrib for propagator-only additions where a full crate is a bit heavy, but as this will likely end up having both, a new crate makes sense.

**Comment by hdost** [comment]
We've discussed recently in the SIG #841 and the current idea is that for now we can accept things into the repo, though likely in a separate crate for things like these extensions. Once stable we'd definitely move into a contrib.

**Comment by djc** [comment]
@hdost, I'm confused, this is effectively what opentelemetry-stackdriver provides today.

**Comment by hdost** [comment]
> @hdost, I'm confused, this is effectively what opentelemetry-stackdriver provides today.
> 
> 
> 
> Edit: ahh, this is specifically about a propagator.

Indeed but it seems feasible that it could be a feature of the stackdriver crate but it's conceivable you wouldn't want both parts.
